crates/cast/src/cmd/run.rs

```diff
@@ -155,6 +155,19 @@ impl RunArgs {
         // fetch the block the transaction was mined in
         let block = provider.get_block(tx_block_number.into()).full().await?;
 
+        // Only fetch raw block data if zk_force is enabled
+        let mut raw_block = None;
+        if self.zk_force {
+            raw_block = Some(
+                provider
+                    .raw_request::<_, serde_json::Value>(
```

Comment:
I believe if you pass `Vec<zksync_types::Transaction>` instead of serde_json it will automatically try to decode it